web/src/people.ts

```diff
@@ -1501,6 +1518,15 @@ export function extract_people_from_message(message: MessageWithBooleans): void
 
         report_late_add(user_id, person.email);
 
+        if (!settings_data.user_can_access_all_other_users()) {
+            // We do not want to add inaccessible users to the users
+            // data as we do not want to show them in right sidebar.
+            // This may lead to cases where an accessible user is
+            // not present in the users data for some time, but
+            // it will eventually be added later.
+            continue;
```

Comment:
I am not sure about this approach, but we do not have any way to know whether the user ID not present in users data is invalid or is for a unknown user or just a accessible user for which the client does not have any data.

Reply:
This is probably the wrong fix -- we need to call `_add_user` even without this feature just to avoid throwing unknown user exceptions.

Probably we don't need a change here at all, and instead we should just have `make_user` set a `inaccessible_user: bool` property that it's an "unknown user", and have the right sidebar logic just exclude such fake users in `filter_user_ids`, where we filter muted users and bot users.